src/java.base/share/classes/jdk/internal/util/ModifiedUtf.java

```diff
@@ -33,6 +33,9 @@
  * @since 24
  */
 public abstract class ModifiedUtf {
+    //Max length in Modified UTF-8 bytes for class names.(see max_symbol_length in symbol.hpp)
+    public static final int JAVA_CLASSNAME_MAX_LEN = 65535;
```

Comment:
max_symbol_length is not just class names - it is presumably the limit for modified UTF-8, as seen in `java.io.DataOutput::writeUTF`. We can just use a more generic name like `MAX_ENCODED_LENGTH`.

Reply:
These are specific to the encoding of all UTF-8 Class File constant too, instead of being Class specific.